cogs/source/source.py

```diff
@@ -0,0 +1,146 @@
+from redbot.core import commands
+import tracemoepy
+from tracemoepy.errors import TooManyRequests
+from tracemoepy.errors import EntityTooLarge
+from tracemoepy.errors import ServerError
+from tracemoepy.errors import EmptyImage
+import math
+from discord import Embed
+
+
+class Source(commands.Cog):
+    """Looks up source of anime and pictures, may get NSFW results. Requires tracemoepy to be installed"""
+
+    """ run 'pip install tracemoepy' on your redbot virtual environment to install"""
+
+    # @commands.command()
+    @commands.group(name="source", invoke_without_command=True)
+    async def sourceCommand(self, ctx):
+        """
+        Looks for source of image
+
+        Parameters:
+        -----------
+        make sure to attach a png or jpg image, or type 'source url URL_HERE'
+        """
+
+        helpmsg = "Looks for source of image\nMake sure to attach a png or jpg image, or type 'source url URL_HERE'"
+        tracemoe = tracemoepy.tracemoe.TraceMoe()
+        try:
+            await ctx.trigger_typing()
+            try:
+                attachment = ctx.message.attachments[0].url
+            except:
+                # await ctx.send(helpmsg)
+                # await self.bot.send_help_for(self, command="source")
+                await ctx.send_help(command="source")
+                return
+            result = tracemoe.search(attachment.strip("<>"), is_url=True)
+            titleEnglish = f"{result.docs[0].title_english}"
+            anilistID = f"{result.docs[0].anilist_id}"
+            episode = f"{result.docs[0].episode}"
+            similarity = float(f"{result.docs[0].similarity}")
+            URL = "https://anilist.co/anime/" + anilistID
+
+            # embed = Embed(title=titleEnglish, url=URL)
+            # embed.add_field(name="Episode", value=episode)
+            # embed.add_field(name="Similarity", value=('%.3f'%((similarity)*100)))
+            if similarity < 0.8:
+                await ctx.send(
+                    "Anime: "
+                    + titleEnglish
+                    + "\nEpisode: "
+                    + episode
+                    + "\nWARNING: Similarity less than 80%, result may not be accurate"
+                    + "\n"
+                    + URL
+                )
+                # embed.add_field(name="Warning", value="Similarity less than  80%, results may be innacurate")
+                #
+            else:
+                await ctx.send(
+                    "Anime: "
+                    + titleEnglish
+                    + "\nSimilarity: "
+                    + ("%.3f" % ((similarity) * 100))
+                    + "%"
+                    + "\nEpisode: "
+                    + episode
+                    + "\n"
+                    + URL
+                )
+                # await ctx.send(embed=embed)
+        except TooManyRequests:
+            await ctx.send("Too many requests sent")
```

Comment:
I think something like "Please try again later" would be better from a user's point of view.